cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_association.spec.js

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { MiscUtils } from '@opensearch-dashboards-test/opensearch-dashboards-test-library';
+const miscUtils = new MiscUtils(cy);
+const workspaceName = 'test_workspace_collaborators';
+let workspaceId;
+let dataSourceTitle1 = 'no_auth_data_source_title_1';
+let dataSourceTitle2 = 'no_auth_data_source_title_2';
+let dataSourceId1;
+let dataSourceId2;
+if (
+  Cypress.env('WORKSPACE_ENABLED') &&
+  Cypress.env('SECURITY_ENABLED') &&
+  Cypress.env('DATASOURCE_MANAGEMENT_ENABLED')
+) {
+  describe('Workspace association data source', () => {
+    before(() => {
+      cy.createDataSourceNoAuth({ title: dataSourceTitle1 }).then((result) => {
+        dataSourceId1 = result[0];
+      });
+      cy.createDataSourceNoAuth({ title: dataSourceTitle2 }).then((result) => {
+        dataSourceId2 = result[0];
+      });
+    });
+    beforeEach(() => {
+      cy.deleteWorkspaceByName(workspaceName);
+      //Create a workspace before each test
+      cy.createWorkspace({
+        name: workspaceName,
+        features: ['use-case-observability'],
+        settings: {
+          permissions: {
+            library_write: { users: ['%me%'] },
+            write: { users: ['%me%'] },
+          },
+        },
+      }).then((value) => {
+        workspaceId = value;
+      });
+    });
+
+    after(() => {
+      cy.deleteDataSource(dataSourceId1);
+      cy.deleteDataSource(dataSourceId2);
+    });
+    afterEach(() => {
+      cy.deleteWorkspaceById(workspaceId);
+    });
+
+    it('should associate and dissociate data source successfully', () => {
```

Comment:
Do we need to add test cases for associating DQC?

Reply:
We don't have related infra of this type data source for now, may need to figure out a way to create this type data source first.

Reply:
We don't have related infra of this type data source for now, may need to figure out a way to create this type data source with related team first.